lib/cloudinary/base_config.rb

```diff
@@ -0,0 +1,82 @@
+module Cloudinary
+  class BaseConfig
```

Comment:
In my IDE (RubyMine) all config properties are underlined with warning, saying: 
`Cannot find [property name]`. 
Those warnings need to be fixed.

Overall a very good improvement!